core/node/node_framework/src/implementations/layers/eth_sender.rs

```diff
@@ -0,0 +1,174 @@
+use std::sync::Arc;
+
+use zksync_config::configs::{
+    chain::NetworkConfig,
+    eth_sender::{ETHSenderConfig, SenderConfig},
+    ContractsConfig, ETHClientConfig,
+};
+use zksync_core::{
+    eth_sender::{Aggregator, EthTxAggregator, EthTxManager},
+    l1_gas_price::L1TxParamsProvider,
+};
+use zksync_dal::ConnectionPool;
+use zksync_eth_client::{clients::PKSigningClient, BoundEthInterface};
+use zksync_types::{Address, L2ChainId};
+
+use crate::{
+    implementations::resources::{
+        eth_interface::BoundEthInterfaceResource, l1_tx_params::L1TxParamsResource,
+        object_store::ObjectStoreResource, pools::MasterPoolResource,
+    },
+    service::{ServiceContext, StopReceiver},
+    task::Task,
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+#[derive(Debug)]
+pub struct EthSenderLayer {
+    eth_sender_config: ETHSenderConfig,
+    contracts_config: ContractsConfig,
+    eth_client_config: ETHClientConfig,
+    network_config: NetworkConfig,
+}
+
+impl EthSenderLayer {
+    pub fn new(
+        eth_sender_config: ETHSenderConfig,
+        contracts_config: ContractsConfig,
+        eth_client_config: ETHClientConfig,
+        network_config: NetworkConfig,
+    ) -> Self {
+        Self {
+            eth_sender_config,
+            contracts_config,
+            eth_client_config,
+            network_config,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl WiringLayer for EthSenderLayer {
+    fn layer_name(&self) -> &'static str {
+        "eth_sender_layer"
+    }
+
+    async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
+        // Get resources
+        let pool_resource = context.get_resource::<MasterPoolResource>().await?;
+        let pool = pool_resource.get().await.unwrap();
+
+        let eth_client_resource = context.get_resource::<BoundEthInterfaceResource>().await?;
+        let eth_client = eth_client_resource.0;
+
+        let object_store_resource = context.get_resource::<ObjectStoreResource>().await?;
+        let object_store = object_store_resource.0;
+
+        // Create and add tasks
+        let eth_client_blobs = PKSigningClient::from_config_blobs(
+            &self.eth_sender_config,
+            &self.contracts_config,
+            &self.eth_client_config,
+        );
+        let eth_client_blobs_addr = eth_client_blobs.clone().map(|k| k.sender_account());
```

Comment:
Side note: this isn't quite idiomatic for the framework, as generally we'd expect anything that may have alternative implementations to be expressed as a resource.
However, the approach chosen in eth sender is not quite idiomatic itself (having 2nd PKSigningClient as an `Option` with bunch of `if` statements is somewhat clunky), so it doesn't make much sense to solve it on the framework side.

Let's keep it as you wrote for now, but I created [a task to handle this](https://linear.app/matterlabs/issue/QIT-28/framework-handle-second-pksigningclient-for-eth-sender), I'll work on it a bit later.

This comment is just FYI, no action required :)